content/guides/hot-module-replacement.md

```diff
@@ -18,125 +19,194 @@ related:
++ main.js:4395 [WDS] App hot update...
++ main.js:4330 [HMR] Checking for updates on the server...
++ main.js:10024 Accepting the updated printMe module!
++ 0.4b8ee77….hot-update.js:10 Updating print.js...
++ main.js:4330 [HMR] Updated modules:
++ main.js:4330 [HMR]  - 20
++ main.js:4330 [HMR] Consider using the NamedModulesPlugin for module names.
 ```
 
-
 ## Gotchas
 
-Hot Module Replacement can be tricky. For example, let's say I have the following class:
+Hot Module Replacement can be tricky. To show this, let's go back to our working example. If you go ahead and click the button on the example page, you will realize the console is printing the old `printMe` function.
+
+This is happening because the button's `onclick` event handler is still bind to the original `printMe` function.
+
+To make this work with HMR we need to update that binding to the new `printMe` function using `module.hot.accept`
 
 ``` js
-class Logger {
-  log(text) {
-    console.log('Logging some text: ', text)
-  }
+import _ from 'lodash';
+import printMe from './print.js';
+
+if (module.hot) {
+    module.hot.accept('./print.js', function() {
+    console.log('Accepting the updated printMe module!');
+    document.body.removeChild(element);
+    element = component(); // update binding now that new printMe module has been acepted
+    document.body.appendChild(element);
+  })
 }
-```
 
-Even if the underlying module containing this class is patched with new code, any existing instances of the class still have the old `log` method. Meaning if we changed what that method does, it wouldn't be reflected in those old instances unless we re-instantiate them somehow using `module.hot.accept`.
+let element = component();
 
-This is just one example, but there are many others that can easily trip people up. Luckily, there are a lot of loaders out there, some mentioned below, that will make using this process much easier.
+function component() {
+  var element = document.createElement('div');
+  var btn = document.createElement('button');
 
+  element.innerHTML = _.join(['Hello', 'webpack'], ' ');
 
-## HMR with Stylesheets
+  btn.innerHTML = 'Click me and check the console!';
+  btn.onclick = printMe;  // onclick event is bind to the original printMe function
 
-We can use the `style-loader` to achieve Hot Module Replacement with CSS. This loader uses `module.hot.accept` behind the scenes to patch `<style>` tags when CSS dependencies are updated. So, with the following webpack configuration...
+  element.appendChild(btn);
 
-``` js
-module.exports = {
-  // ...
-  module: {
-    rules: [
-      {
-        test: /\.css$/,
-        use: [ 'style-loader', 'css-loader' ]
-      }
-    ]
-  },
-  // ...
+  return element;
 }
+
+document.body.appendChild(element);
+```
+
+This is just one example, but there are many others that can easily trip people up. Luckily, there are a lot of loaders out there, some mentioned below, that will make using this process much easier.
+
```

Comment:
Add an extra line break between sections (similar to my other `h2` comment above).